src/core/reducer.ts

```diff
@@ -121,6 +128,79 @@ function initializeDeltalog(
   };
 }
 
+/**
+ * ExtractTransientsFromState
+ *
+ * Split out transients from the a TransientState
+ */
+function ExtractTransients(
+  transientState: TransientState | null
+): [State | null, TransientMetadata | null] {
```

Comment:
Missed this first time round, but the actual return type here is:

```suggestion
): [State | null, TransientMetadata | undefined] {
```